File: spring-boot/jwt/src/main/java/com/example/jwt/_common/config/WebSecurityConfig.java
package com.example.jwt._common.config;

import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityCustomizer;
import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
import org.springframework.security.config.http.SessionCreationPolicy;
import org.springframework.security.web.SecurityFilterChain;
import org.springframework.security.web.util.matcher.AntPathRequestMatcher;

@Configuration
@EnableWebSecurity
@RequiredArgsConstructor
public class WebSecurityConfig {

    @Bean
    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
        return http
                .csrf(AbstractHttpConfigurer::disable)
                .formLogin(AbstractHttpConfigurer::disable)
                .rememberMe(AbstractHttpConfigurer::disable)
                .logout(AbstractHttpConfigurer::disable)
                .httpBasic(AbstractHttpConfigurer::disable)

                .sessionManagement(sessionManagement ->
                        sessionManagement.sessionCreationPolicy(SessionCreationPolicy.STATELESS)
                )

                .authorizeHttpRequests(authorizeHttpRequests ->
                        authorizeHttpRequests
                                .requestMatchers("/health").permitAll()
                                .requestMatchers("/auth/sign-in").permitAll()
                                .requestMatchers("/auth/sign-up").permitAll()
                                .requestMatchers("/auth/refresh").permitAll()
                                .requestMatchers("/**").hasAnyRole("USER", "ADMIN")
                                .anyRequest().authenticated()
                )
//                .exceptionHandling(exceptionHandling ->
//                                exceptionHandling.authenticationEntryPoint(authenticationEntryPointImpl)
//                        accessDeniedHandler(accessDeniedHandlerImpl)
//                )
//                .addFilterBefore(
//                        JwtAuthenticationFilter(tokenProvider),
//                        UsernamePasswordAuthenticationFilter::class.java
                .build();
    }

    @Bean
    public WebSecurityCustomizer webSecurityCustomizer() {
        return web -> {
            web.ignoring().requestMatchers(new AntPathRequestMatcher("/h2-console/**"));
            web.ignoring().requestMatchers(new AntPathRequestMatcher("/docs/**"));
        };
    }
}

File: spring-boot-kotlin/jwt/src/main/kotlin/com/example/jwt/security/WebSecurityConfig.kt
package com.example.jwt.security

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.http.SessionCreationPolicy
import org.springframework.security.web.SecurityFilterChain

@Configuration
@EnableWebSecurity
class WebSecurityConfig {
    @Bean
    fun filterChain(http: HttpSecurity): SecurityFilterChain {
        http
            .httpBasic().disable() // rest api 만을 고려하여 기본 설정은 해제하겠습니다.
            .csrf().disable() // csrf 보안 토큰 disable 처리.
            .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS) // 토큰 기반 인증이므로 세션 역시 사용하지 않습니다.
            .and()
            .authorizeHttpRequests()
            .requestMatchers("/**").permitAll()
//            .requestMatchers("/h2-console/**").permitAll()
//            .requestMatchers("/user/**").permitAll()
//            .requestMatchers("/user/**").hasAuthority("ROLE_USER")
        return http.build()
    }
}